### ISSUE TITLE
Opt-in to which instances toot.cat is federated

### ISSUE THREAD

**Issue by polymerwitch** [TootCat/mastodon#42]
Instead of federating with instances by default this would create an Allowed List of instances with which we can update overtime.

* * * *

- [ ] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [ ] This bug happens on a [tagged release](https://github.com/tootsuite/mastodon/releases) and not on `master` (If you're a user, don't worry about this).

**Comment by lynnco** [comment]
Did you get a look at the code awoo.space uses for this ?

**Comment by polymerwitch** [comment]
@lynnco yes, there is some interesting stuff here: https://github.com/phildini/mastodon/tree/private-mastodon

it includes things like only letting authed users see public toots and profiles

**Comment by bit-mancer** [comment]
I like this! Auto-federation is likely going to end up being an issue if any groups ever commit to sustained harassment of the platform...

Also +1 on requiring auth -- in the context of the search discussion today, there's currently nothing stopping someone from bypassing federation entirely and scrapping the .atom feeds to get every non-direct message if they are so inclined.

**Comment by wavebeem** [comment]
I think this is probably a good idea. I can see how some people might be upset at this being exclusionary towards new instances, but honestly I think the pros outweigh the cons here?
